reconcile/openshift_groups.py

```diff
@@ -112,41 +101,48 @@ def fetch_current_state(thread_pool_size, internal, use_jump_host):
     return oc_map, current_state, ocm_clusters
 
 
-def fetch_desired_state(oc_map, enforced_user_keys=None):
+def fetch_desired_state(
+    oc_map: ClusterMap, enforced_user_keys: Optional[list[str]] = None
+) -> list[dict[str, str]]:
     gqlapi = gql.get_api()
-    roles = expiration.filter(gqlapi.query(ROLES_QUERY)["roles"])
-    desired_state = []
+    roles = query_managed_roles(query_func=gqlapi.query).roles or []
+    desired_state: list[dict[str, str]] = []
 
     for r in roles:
-        for a in r["access"] or []:
-            if None in [a["cluster"], a["group"]]:
+        for a in r.access or []:
+            if not a.cluster or not a.group:
                 continue
-            if oc_map and a["cluster"]["name"] not in oc_map.clusters():
+            if a.cluster.name not in oc_map.clusters():
                 continue
 
             user_keys = ob.determine_user_keys_for_access(
-                a["cluster"]["name"],
-                a["cluster"]["auth"],
+                a.cluster.name,
+                a.cluster.auth,
                 enforced_user_keys=enforced_user_keys,
             )
-            for u in r["users"]:
-                for username in {u[user_key] for user_key in user_keys}:
+            for u in r.users:
+                for username in {
+                    u.dict(by_alias=True)[user_key] for user_key in user_keys
```

Comment:
Please use `getattr(u, user_key)` instead of `u.dict(by_alias=True)[user_key]`. It would be much easier to read.